lib/bsdlib/nrf91_sockets.c

```diff
@@ -478,24 +478,45 @@ static int nrf91_socket_offload_accept(int sd, struct sockaddr *addr,
 				       socklen_t *addrlen)
 {
 	int retval;
+	struct nrf_sockaddr *nrf_addr_ptr = NULL;
+	nrf_socklen_t *nrf_addrlen_ptr = NULL;
+	/* Use `struct nrf_sockaddr_in6` to fit both, IPv4 and IPv6 */
+	struct nrf_sockaddr_in6 nrf_addr;
+	nrf_socklen_t nrf_addrlen;
+
+	if ((addr != NULL) && (addrlen != NULL)) {
+		nrf_addr_ptr = (struct nrf_sockaddr *)&nrf_addr;
+		nrf_addrlen_ptr = &nrf_addrlen;
+
+		/* Workaround for the bsdlib issue, making `nrf_accept` to
+		 * expect `nrf_addrlen` to be exactly of
+		 * sizeof(struct nrf_sockaddr_in) size for IPv4
+		 */
+		if (*addrlen == sizeof(struct sockaddr_in)) {
+			nrf_addrlen = sizeof(struct nrf_sockaddr_in);
+		} else {
+			nrf_addrlen = sizeof(struct nrf_sockaddr_in6);
+		}
+	}
 
-	struct nrf_sockaddr nrf_addr;
-	nrf_socklen_t nrf_addrlen = sizeof(nrf_addr);
-
-	retval = nrf_accept(sd, &nrf_addr, &nrf_addrlen);
+	retval = nrf_accept(sd, nrf_addr_ptr, nrf_addrlen_ptr);
```

Comment:
If above check fails, here both nrf_addr_ptr and nrf_addrlen_ptr are NULL. Will BSDlib allocate inside?